### PR TITLE
add smoke tests to run in travis 

### PR DESCRIPTION
**Pull request by kimberlythegeek**
added tests to check default config options are set as expected, and custom values are set as expected

### CODE REVIEW

File: moziris/test/conftest.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#     ):
+#
+#         test = arg_parser.get_core_args()
+#         print("\n\n\test args:\n")
+#         pprint(test.__dict__)
+#         yield test
+#
+#         sys.argv = [sys.argv[0]]
+#
+#         # return arg_parser.get_core_args()
```
Comment (mwobensmith) [inline moziris/test/conftest.py]: Any reason why this line is commented out?

Comment (kimberlythegeek) [reply moziris/test/conftest.py]: Committed this by accident in the first commit, the file has been removed.

File: .travis.yml [hunk 2]
```diff
@@ -11,22 +11,25 @@ install:
...
-    - stage: Ubuntu 16.04 Image search and mouse movement tests
+    - stage: Build
       os: linux
       dist: xenial
       script: pipenv run iris -h
+    - stage: Smoke Tests
+      install: pip install tox
+      script:
+        - tox -e default_config
+        - tox -e custom_config
```
Comment (mwobensmith) [inline .travis.yml]: What I see here are unit tests, written against code that is run in a context outside of the project. I think we should rename this stage to "Unit Tests" and move it before the "Build" stage, as it is more low-level and errors here should prevent us from going any further in the validation process. Do you agree?

Also, the "Build" stage could be followed by a "Smoke Tests" stage that invokes a script that actually performs integration scenarios. Much like how the "Build" stage currently just invokes the -h flag, a "Smoke Tests" stage could run commands such as `pipenv run iris sample -t specific_test <args>` and confirm specific behavior. However, we can cover that later.

File: moziris/test/conftest.py (hunk 1 above)
Comment (mwobensmith) [reply moziris/test/conftest.py]: Sorry, my mistake.

File: .travis.yml (hunk 2 above)
Comment (kimberlythegeek) [reply .travis.yml]: I really wouldn't consider these unit tests. The only piece of iris core code it touches is the ability to set options. They do a low level check to ensure the tests will be able to run as expected.

I see your point about the order of the stages. the Build stage takes a while so if Smoke will fail we should fail faster. I've just pushed those changes to my fork and the travis builds are passing.

I could add more smoke tests for sure, using the `sample` target

Comment (mwobensmith) [reply .travis.yml]: [Unit test](https://en.wikipedia.org/wiki/Unit_testing) vs [smoke test.](http://softwaretestingfundamentals.com/smoke-testing/)

I think the tests you've written fit the description of unit tests, but do not verify a successful build, which is why they shouldn't be considered smoke tests either. I think they are useful, and if we'd like to write other low-level tests like this, let's run them together before the "Build" step.

We have to build before we can run smoke tests, if we are to follow the definition.

**Comment by kimberlythegeek** [comment]
It's worth noting that I am unable to run both the default and custom config tests in the same tox environment. it seems that once the configuration is set in iris it is unable to be set again. I tried digging into that a bit, but so far have not found a solution. for now I will keep them separate.

**Pull request merged**